lib/Common/Codex/Utf8Codex.cpp

```diff
@@ -377,7 +377,7 @@ namespace utf8
             return ptr;
     }
 
-    void DecodeInto(__out_ecount_full(cch) char16 *buffer, LPCUTF8 ptr, size_t cch, DecodeOptions options)
+    void DecodeInto(__out_ecount_full(cch) char16 *buffer, LPCUTF8 ptr, LPCUTF8 end, size_t cch, DecodeOptions options)
```

Comment:
These two sets are closer but not duplicates. DecodeUnitsInto advances the source pointer and the fast path for reading characters is a little different.